tests/test_preprocess.py

```diff
@@ -124,3 +132,44 @@ def test_extend_lines(self):
         ext5 = mm.extend_lines(gdf, 2)
         assert ext5.length.sum() > gdf.length.sum()
         assert ext5.length.sum() == pytest.approx(6.2, rel=1e-3)
+
+    def test_roundabout_simplification_1(self):
+        if not GPD_09:
+            with pytest.raises(ImportError):
+                mm.roundabout_simplification(self.df_streets_rabs)
```

Comment:
See the comment on decorators below. What I would do is to turn this into a test by itself.

```py
@pytest.mark.skipif(GPD_09, reason="requires geopandas 0.9+")
def test_roundabout_simplification_gpd_error(self):
    with pytest.raises(ImportError, match="`roundabout_simplification` requires geopandas 0.9.0"):
        mm.roundabout_simplification(self.df_streets_rabs)
```

This is to make sure each test focuses on one thing to check, in this case an error. I am also adding `match` to check that the exact error is raised, not just some ImportError.